### PR TITLE
Save system

### LINKED ISSUE #17: Save/Load System

### PR DESCRIPTION
**Pull request by Kainkun**
close #17  currently only saves and load things that are already in the scene. it cant instantiate or destroy yet

### CODE REVIEW

**Comment by BUNN1E5** [comment]
Omg this is a lot

File: Assets/Scripts/SaveSystem/testClasses/TestPlayer.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+
+
+
+
+
+        #region SAVE SYSTEM
+        [System.Serializable]
+        protected class TestPlayerSaveData : TestEntitySaveData //inherited class that is a container for data that will be saved
```
Comment (BUNN1E5) [inline Assets/Scripts/SaveSystem/testClasses/TestPlayer.cs]: just confirming we need to implement this for all entities?

**Review by BUNN1E5 — APPROVED** [review]
Other than really pedantic things like "you shouldn't push test scripts" Looks fine by me,

File: Assets/Scripts/SaveSystem/SaveableGameObject.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using UnityEngine.SceneManagement;
+
+namespace SaveSystem
+{
+    //Component that is added to GameObjects that have other components (that implements ISaveable) that need to be saved
+    [ExecuteInEditMode]
+    public class SaveableGameObject : MonoBehaviour
```
Comment (BUNN1E5) [inline Assets/Scripts/SaveSystem/SaveableGameObject.cs:11]: Wait so you need to add this component to the gameobject to be saved?

File: Assets/Scripts/SaveSystem/testClasses/TestEntity.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+
+
+
+
+
+        #region SAVE SYSTEM
+        [System.Serializable]
+        protected class TestEntitySaveData : SaveData //class that is a container for data that will be saved
```
Comment (BUNN1E5) [inline Assets/Scripts/SaveSystem/testClasses/TestEntity.cs]: So each ISavableComponent needs to create a subclass with SaveData?

File: Assets/Scripts/SaveSystem/testClasses/TestPlayer.cs (hunk 1 above)
Comment (Kainkun) [reply Assets/Scripts/SaveSystem/testClasses/TestPlayer.cs]: well, it needs to be implemented for all components that need to be saved

Comment (Kainkun) [reply Assets/Scripts/SaveSystem/testClasses/TestPlayer.cs]: i was wondering if theres a better way to do this though. maybe adding an attribute to variables you want to save. and automatically handling it that way

File: Assets/Scripts/SaveSystem/SaveableGameObject.cs (hunk 2 above)
Comment (Kainkun) [reply Assets/Scripts/SaveSystem/SaveableGameObject.cs:11]: yes

Comment (Kainkun) [reply Assets/Scripts/SaveSystem/SaveableGameObject.cs:11]: that searches the game object for all saveable components/variables

File: Assets/Scripts/SaveSystem/testClasses/TestEntity.cs (hunk 3 above)
Comment (Kainkun) [reply Assets/Scripts/SaveSystem/testClasses/TestEntity.cs]: yes, i wonder if theres a better way though

**Pull request merged**